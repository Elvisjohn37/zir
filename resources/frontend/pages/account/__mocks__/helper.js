let mockedModule = jest.createMockFromModule('../helper.js');

mockedModule.openAccountWindow = jest.fn();

module.exports = mockedModule;
