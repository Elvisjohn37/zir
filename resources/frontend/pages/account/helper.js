export function openAccountWindow(url) {
	window.open(url, 'account_window', 'width = 670, height = 800');
}
