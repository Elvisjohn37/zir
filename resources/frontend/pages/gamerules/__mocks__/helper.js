let mockedModule = jest.createMockFromModule('../helper');

mockedModule.extractProductGameType = (gameRulesList) => {
	let extractedItems = [];
	let indexMap = {};
	let nextIndex = 0;

	gameRulesList.forEach((item) => {
		if (item.productName in indexMap) {
			if (!extractedItems[indexMap[item.productName]].gameTypes.includes(item.gameTypeName)) {
				extractedItems[indexMap[item.productName]].gameTypes.push(item.gameTypeName);
			}
		} else {
			indexMap[item.productName] = nextIndex;
			nextIndex++;

			extractedItems.push({
				productName: item.productName,
				gameTypes: [item.gameTypeName],
			});
		}
	});

	return extractedItems;
};

module.exports = mockedModule;
