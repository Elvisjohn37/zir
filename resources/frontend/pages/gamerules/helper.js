import lodash from 'lodash';

function extractProductGameType(gameRulesList) {
	let extractedItems = [];
	let indexMap = {};
	let nextIndex = 0;

	gameRulesList.forEach((item) => {
		if (item.productName in indexMap) {
			if (!lodash.includes(extractedItems[indexMap[item.productName]].gameTypes, item.gameTypeName)) {
				extractedItems[indexMap[item.productName]].gameTypes.push(item.gameTypeName);
			}
		} else {
			indexMap[item.productName] = nextIndex;
			nextIndex++;

			extractedItems.push({
				productName: item.productName,
				gameTypes: [item.gameTypeName],
			});
		}
	});

	return extractedItems;
}

export { extractProductGameType };
