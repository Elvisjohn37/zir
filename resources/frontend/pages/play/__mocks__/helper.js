let postMessageHandler = () => jest.fn();

let setPostMessageHandler = () => jest.fn();

let resetPostMessageHandler = () => jest.fn();

resetPostMessageHandler();

// _________________________________________

let windowResponseData,
	isMobile = false;

let windowResponse = () => windowResponseData;

let resetWindowResponse = () => {
	windowResponseData = {
		result: true,
		data: {
			url: 'https://eyecon.classicku.com/launch/lek?uid=ANqWb0TXAFr1%252BZT7aQmY%252F1M%253D&guid=124275588462ff2e5f3973e3.1098258&gameid=71656&brand=LEK&cur=ZWD&lang=en&homeURL=https%3A%2F%2Fgames.classicku.com%2Fclosegame%3FgameID%3DDVbr%26clientID%3DML641',
			title: 'Beat The Bobbies',
			maxPayout: '700000000.000000',
			isMobile: false,
			viewport: 'desktop',
			productID: 'zy',
			balanceInfo: { playableBalance: '1457.800000', currencyCode: 'ZWD' },
		},
	};
};

let setWindowResponse = (data) => {
	windowResponseData = {
		...windowResponseData,
		data: {
			...windowResponseData.data,
			...data,
		},
	};
};

let setErrorResponse = () => {
	windowResponseData = {
		...windowResponseData,
		result: false,
	};
};

resetWindowResponse();

// _________________________________________

module.exports = { postMessageHandler, windowResponse, setWindowResponse, resetWindowResponse, setErrorResponse };
