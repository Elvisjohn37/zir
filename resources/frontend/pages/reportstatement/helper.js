import { getStatementReport as getStatementReportAjax } from 'frontend/ajax/backend';

export function getStatementReport(
	isMounted,
	setDateRange,
	statementRange,
	setStatementData,
	setStatementDataLoading,
	setStatementDataError
) {
	setStatementDataLoading(true);

	getStatementReportAjax(
		statementRange,
		!isMounted,
		(response) => {
			setStatementDataLoading(false);
			setStatementData({
				content: response.data.data.content,
				footer: response.data.data.footer,
			});
			if (!isMounted) {
				setDateRange({
					isLoading: false,
					range: response.data.data.dateRange,
				});
			}
		},
		() => {
			setStatementDataError(true);
			setStatementDataLoading(false);
		}
	);
}
