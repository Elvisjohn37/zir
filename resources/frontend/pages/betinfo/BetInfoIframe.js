import React, { useEffect, useState, useRef } from 'react';
import Spinner from 'frontend/components/Spinner';
import styles from './BetInfoIframe.module';
import SeamlessIframe from 'frontend/components/SeamlessIframe';
import { windowResponse } from 'frontend/utils/helper';
import classnames from 'classnames';

const IFRAME_SIZE = {
	eyecon: {
		width: 400,
		height: 600,
	},

	funky: {
		width: 387,
		height: 635,
	},
};

const WINDOW_SIZE = {
	eyecon: {
		width: 410,
		height: 605,
	},
	funky: {
		width: 400,
		height: 700,
	},
};

export default function BetInfoIframe({ type }) {
	let iframeRef = useRef();
	let response = windowResponse();
	let [isLoading, setIsLoading] = useState(true);

	useEffect(() => {
		window.resizeTo(WINDOW_SIZE[type].width, WINDOW_SIZE[type].height);
	});

	return (
		<div className={classnames([styles.betInfoIframe, styles[type]])}>
			{isLoading && <Spinner color="primary" className={styles.iframeSpinner} />}
			<SeamlessIframe
				width={IFRAME_SIZE[type].width}
				height={IFRAME_SIZE[type].height}
				scrolling="1"
				src={response.data.data.url}
				ref={iframeRef}
				onLoad={() => setIsLoading(false)}
			></SeamlessIframe>
		</div>
	);
}
