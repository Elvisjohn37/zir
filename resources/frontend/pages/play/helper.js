import { windowResponse } from 'frontend/utils/helper';

let postMessageHandlerWrapper = () => {
	let allowedOrigin = windowResponse() ? windowResponse().data.url : '';

	return function ({ data, origin }) {
		let originUrl = new URL(origin);
		let gameUrl = new URL(allowedOrigin);

		if ((gameUrl.origin == originUrl.origin || window.location.host == originUrl.host) && data == 'closeGame') {
			window.location.href = '/';
		}
	};
};

let postMessageHandler = postMessageHandlerWrapper();

export { postMessageHandler, windowResponse };
