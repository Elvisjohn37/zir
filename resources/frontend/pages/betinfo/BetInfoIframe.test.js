import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import BetInfoIframe from './BetInfoIframe';
import { setWindowResponse } from 'frontend/utils/helper';

jest.mock('frontend/components/Spinner');
jest.mock('frontend/components/SeamlessIframe');
jest.mock('frontend/utils/helper');
jest.mock('classnames');

let windowResize = jest.fn();
window.resizeTo = windowResize;

describe('</BetInfoIframe>', () => {
	let sampleGameUrl = 'www.game.com';

	beforeEach(() => {
		setWindowResponse({
			result: true,
			data: {
				data: {
					url: sampleGameUrl,
				},
			},
		});
	});

	it('should match snapshot', () => {
		let { asFragment, getByTestId } = render(<BetInfoIframe type="eyecon" />);
		fireEvent.load(getByTestId('SeamlessIframe'));
		expect(asFragment()).toMatchSnapshot();
	});

	it('should display loading indicator when bet details iframe is not yet ready', () => {
		let { getByTestId } = render(<BetInfoIframe type="eyecon" />);
		expect(getByTestId('Spinner')).toBeInTheDocument(1);
	});

	it('should display eyecon bet details iframe', () => {
		let { getByTestId } = render(<BetInfoIframe type="eyecon" />);
		let iframe = getByTestId('SeamlessIframe');

		expect(iframe).toBeInTheDocument();
		expect(iframe).toHaveAttribute('width', '400');
		expect(iframe).toHaveAttribute('height', '600');
		expect(iframe).toHaveAttribute('src', sampleGameUrl);
		expect(windowResize).toHaveBeenCalledWith(410, 605);
	});

	it('should display funky bet details iframe', () => {
		let { getByTestId } = render(<BetInfoIframe type="funky" />);
		let iframe = getByTestId('SeamlessIframe');

		expect(iframe).toBeInTheDocument();
		expect(iframe).toHaveAttribute('width', '387');
		expect(iframe).toHaveAttribute('height', '635');
		expect(iframe).toHaveAttribute('src', sampleGameUrl);
		expect(windowResize).toHaveBeenCalledWith(400, 700);
	});
});
