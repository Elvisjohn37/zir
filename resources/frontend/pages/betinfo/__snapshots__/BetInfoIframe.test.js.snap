// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`</BetInfoIframe> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="betInfoIframe eyecon"
  >
    <iframe
      data-testid="SeamlessIframe"
      height="600"
      scrolling="1"
      src="www.game.com"
      width="400"
    />
  </div>
</DocumentFragment>
`;
